### PR TITLE
DOC Auto scaffolding has_one relations

### PR DESCRIPTION
**Pull request by emteknetnz**
Issue https://github.com/silverstripe/silverstripe-admin/issues/1618

I don't know why CI is complaining about linting, it's branched off the latest commit in `5`

### CODE REVIEW

File: en/04_Changelogs/5.2.0.md [hunk 1]
```diff
@@ -157,6 +157,12 @@ Both of these fields include a `setIsLazyLoaded()` method which will load a limi
 
 Note that these are both powered by react components, and are only intended to be used within the CMS. If you want to use them on the front-end of your project you will need to provide your own templates and JavaScript implementation for them.
 
+#### Auto scaffolding of `has_one` relations
+
+A `SearchableDropdownField` will now be used when automatically scaffolding `has_one` relations on a form in the CMS. Previously a `DropdownField` was used, and when there were over 100 items a [`NumericField`](api:SilverStripe\Forms\NumericField) was used which was not user friendly.
+
+```php
```
Comment (ChloeSartorelli) [inline en/04_Changelogs/5.2.0.md]:
```suggestion
```
Pretty sure this is what's making CI angry

File: en/04_Changelogs/5.2.0.md [hunk 2]
```diff
@@ -157,6 +157,12 @@ Both of these fields include a `setIsLazyLoaded()` method which will load a limi
 
 Note that these are both powered by react components, and are only intended to be used within the CMS. If you want to use them on the front-end of your project you will need to provide your own templates and JavaScript implementation for them.
 
+#### Auto scaffolding of `has_one` relations
+
+A `SearchableDropdownField` will now be used when automatically scaffolding `has_one` relations on a form in the CMS. Previously a `DropdownField` was used, and when there were over 100 items a [`NumericField`](api:SilverStripe\Forms\NumericField) was used which was not user friendly.
```
Comment (ChloeSartorelli) [inline en/04_Changelogs/5.2.0.md:162]:
```suggestion
A `SearchableDropdownField` will now be used when automatically scaffolding `has_one` relations on a form in the CMS. Previously a `DropdownField` was used, and when there were over 100 items a [`NumericField`](api:SilverStripe\Forms\NumericField) was used which was not user friendly.
```

Please add a note here about the change to `dropdown_field_threshold` (i.e. it controls the threshold for lazy loading now)

**Review by GuySartorelli — CHANGES_REQUESTED** [review]

File: en/04_Changelogs/5.2.0.md (hunk 2 above)
Comment (emteknetnz) [reply en/04_Changelogs/5.2.0.md:162]: Done

File: en/04_Changelogs/5.2.0.md (hunk 1 above)
Comment (emteknetnz) [reply en/04_Changelogs/5.2.0.md]: Removed

File: en/04_Changelogs/5.2.0.md [hunk 3]
```diff
@@ -157,6 +157,12 @@ Both of these fields include a `setIsLazyLoaded()` method which will load a limi
...
+
+Previously the [`DBForeignKey.dropdown_field_threshold`](api::SilverStripe\ORM\FieldType\DBForeignKey.dropdown_field_threshold) config property was used as the threshold of the number of options to decide when to switch between auto-scaffolding a `DropdownField` and a `NumericField`. This configuration property is now used as the threshold of the number of options to decide when to start using lazy-loading for the `SearchableDropdownField`.
```
Comment (ChloeSartorelli) [inline en/04_Changelogs/5.2.0.md]:
```suggestion
Previously the [`DBForeignKey.dropdown_field_threshold`](api:SilverStripe\ORM\FieldType\DBForeignKey->dropdown_field_threshold) config property was used as the threshold of the number of options to decide when to switch between auto-scaffolding a `DropdownField` and a `NumericField`. This configuration property is now used as the threshold of the number of options to decide when to start using lazy-loading for the `SearchableDropdownField`.
```
The docs search that the api links turn into don't know what to do with `Class.Config` so even though that's how we refer to it in the plain text, we have to treat it as a property in the api link. Both `->` and `::` work here so pick whichever of those you think is nicer.

**Review by GuySartorelli — CHANGES_REQUESTED** [review]

File: en/04_Changelogs/5.2.0.md (hunk 3 above)
Comment (emteknetnz) [reply en/04_Changelogs/5.2.0.md]: That's annoying. I'll use -> 

Updated

Comment (ChloeSartorelli) [reply en/04_Changelogs/5.2.0.md]: Yup lol
A while ago (when I learned this myself) I thought about raising a PR to allow dot syntax in the API link but it just didn't seem worth it in the end.

File: en/04_Changelogs/5.2.0.md [hunk 4]
```diff
@@ -157,6 +157,12 @@ Both of these fields include a `setIsLazyLoaded()` method which will load a limi
 
 Note that these are both powered by react components, and are only intended to be used within the CMS. If you want to use them on the front-end of your project you will need to provide your own templates and JavaScript implementation for them.
 
+#### Auto scaffolding of `has_one` relations
+
+A `SearchableDropdownField` will now be used when automatically scaffolding `has_one` relations on a form in the CMS. Previously a `DropdownField` was used, and when there were over 100 items a [`NumericField`](api:SilverStripe\Forms\NumericField) was used which was not user friendly.
+
+Previously the [`DBForeignKey.dropdown_field_threshold`](api::SilverStripe\ORM\FieldType\DBForeignKey->dropdown_field_threshold) config property was used as the threshold of the number of options to decide when to switch between auto-scaffolding a `DropdownField` and a `NumericField`. This configuration property is now used as the threshold of the number of options to decide when to start using lazy-loading for the `SearchableDropdownField`.
```
Comment (ChloeSartorelli) [inline en/04_Changelogs/5.2.0.md]:
```suggestion
Previously the [`DBForeignKey.dropdown_field_threshold`](api:SilverStripe\ORM\FieldType\DBForeignKey->dropdown_field_threshold) config property was used as the threshold of the number of options to decide when to switch between auto-scaffolding a `DropdownField` and a `NumericField`. This configuration property is now used as the threshold of the number of options to decide when to start using lazy-loading for the `SearchableDropdownField`.
```
Remove extra colon.
This was in https://github.com/silverstripe/developer-docs/pull/425#discussion_r1430677061 as well but got lost in the property syntax stuff.

**Review by GuySartorelli — CHANGES_REQUESTED** [review]

File: en/04_Changelogs/5.2.0.md (hunk 4 above)
Comment (emteknetnz) [reply en/04_Changelogs/5.2.0.md]: Updated

**Review by GuySartorelli — APPROVED** [review]
LGTM

**Pull request merged**